langlib/lang.int/src/main/ballerina/int.bal

```diff
@@ -215,3 +217,39 @@ public isolated function fromHexString(string s) returns int|error = @java:Metho
     'class: "org.ballerinalang.langlib.integer.FromHexString",
     name: "fromHexString"
 } external;
+
+# Returns an iterable object that iterates over a range of integers.
+# The integers returned by the iterator belong to the set S,
+# where S is `{ rangeStart + step*i such that i >= 0 }`.
+# When `step > 0`, the members of S that are `< rangeEnd` are returned in increasing order.
+# When `step < 0`, the members of S that are `> rangeEnd` are returned in decreasing order.
+# When `step = 0`, the function panics.
+# + rangeStart - the first integer to be returned by the iterator
+# + rangeEnd - the exclusive limit on the integers returned by the iterator
+# + step - the difference between successive integers returned by the iterator;
+#    a positive value gives an increasing sequence; a negative value gives
+#    a decreasing sequence
+# + return - an iterable object
+public isolated function range(int rangeStart, int rangeEnd, int step) returns object {
+    *object:Iterable;
+    public isolated function iterator() returns object {
+        public isolated function next() returns record {|
+            int value;
+        |}?;
+    };
+} {
+    int direction = getStepDirection(step);
+    internal:__IntRange intRange = new (rangeStart, rangeEnd + direction, step);
+    return intRange;
+}
+
+isolated function getStepDirection(int step) returns int {
+    if (step == 0) {
+        panic error("step cannot be 0");
```

Comment:
Move this validation to the caller. Having this validation here, causes it to print an additional stack trace when it panics. 

```
error: step cannot be 0
        at ballerina.lang.int.0:getStepDirection(int.bal:248)
           ballerina.lang.int.0:range(int.bal:241)
           test:main(test.bal:21)

``